include/libtorrent/aux_/vector.hpp

```diff
@@ -70,7 +71,38 @@ namespace libtorrent { namespace aux {
 		}
 
 		IndexType end_index() const
-		{ return IndexType(static_cast<underlying_index>(this->size())); }
+		{
+			TORRENT_ASSERT(this->size() <= std::size_t(std::numeric_limits<underlying_index>::max()));
+			return IndexType(static_cast<underlying_index>(this->size()));
+		}
+
+		template <typename U = underlying_index, typename Cond
+			= typename std::enable_if<std::is_signed<U>::value>::type>
+		void resize(underlying_index s)
+		{
+			TORRENT_ASSERT(s >= 0);
+			this->base::resize(std::size_t(s));
+		}
+
+		template <typename U = underlying_index, typename Cond
+			= typename std::enable_if<std::is_signed<U>::value>::type>
+		void resize(underlying_index s, T const& v)
+		{
+			TORRENT_ASSERT(s >= 0);
+			this->base::resize(std::size_t(s), v);
+		}
+
+		void resize(std::size_t s)
+		{
+			TORRENT_ASSERT(s <= std::size_t(std::numeric_limits<underlying_index>::max()));
```

Comment:
since this is in a header, I would expect you would have to do the trick to avoid macro expansion for max. i.e. ``(std::numeric_limits<underlying_index>::max)()``.